tests/test_twisted.py

```diff
@@ -175,12 +176,16 @@ def test_catchesKeyCertificateMismatch(self, tmpdir):
 
 
 class TestForwardCompatibleDHE(object):
-    def test_fakeDHParameterSupport(self, monkeypatch, allFile, recwarn):
+    def test_fakeDHParameterSupport(self, monkeypatch, tmpdir, recwarn):
         """
-        Fake DH parameter support if Twisted doesn't support it.
+        Fake DH parameter support if Twisted doesn't support it for explicitly
+        passed DH parameters.
 
         Warns about deprecation.
         """
+        allFile = tmpdir.join('key_cert_and_chain.pem')
```

Comment:
Since this block appears twice, it should become a fixture.  `keyCertChainFile`?  And maybe rename `allFile` to `KeyCertChainDHFile`?